src/model/databaseModel.ts

```diff
@@ -0,0 +1,38 @@
+import mongoose from "mongoose";
+import { nanoid } from "nanoid";
+
+const shortenedURLSchema = new mongoose.Schema(
+  {
+    customName: {
+      type: String,
+      unique: true,
+      sparse: true,
+    },
+    shortUrl: {
+      type: String,
+      required: false,
+      unique: true,
+      default: () => nanoid().substring(0, 10),
```

Comment:
Nice approach using `default`. 👍🏽 
```suggestion
      default: () => generateCustomName(),
```